dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -1300,6 +1301,10 @@ public void getPreVarPool(TaskInstance taskInstance, Set<String> preTask) {
             }
             if (allProperty.size() > 0) {
                 taskInstance.setVarPool(JSONUtils.toJsonString(allProperty.values()));
+                Map<String, Object> taskParams = JSONUtils.parseObject(taskInstance.getTaskParams(), new TypeReference<Map<String, Object>>() {
+                });
+                taskParams.put("varPool", JSONUtils.toJsonString(allProperty.values()));
```

Comment:
TaskParams and VarPool is different fields, so you can't add varpool into taskParams.

Reply:
org.apache.dolphinscheduler.service.expand.CuringGlobalParams#paramParsingPreparation use taskInstance.varParams to build prepareParamsMap, and prepareParamsMap  was used to replace sql param.But taskInstance.varParams is always null, so I put it.This is why I did it, maybe you have a better idea？

Reply:
I think this is a unified process for all task, so after you fix the spark ui, the custome params can pass to worker.
BTW, the custome params for the current task is localParams.
This is a shell task example, hope it can help you. 
![image](https://user-images.githubusercontent.com/11962619/178404246-ad166c76-1ebf-44e5-9b69-ac5c3be41a1c.png)

Reply:
BTW, I found the flink task also has the same issue in ui.

Reply:
localParams is currently useful, but this is limited to [in] parameters, but for upstream [out] parameters, spark sql currently doesn't work, which is why I use varPool.

Reply:
Now ds just support three task types for parameter transfer, see https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/guide/parameter/context.html
So it's a feature, not a bug, you would better to create another feature issue to do it. WDYT?

Reply:
I will create a feature, but do I need to resubmit a mr about parameter transfer?

Reply:
> I will create a feature, but do I need to resubmit a mr about parameter transfer?

You can submit other pr to the feature issue if you are interested in.